picoquicfirst/picoquicdemo.c

```diff
@@ -276,10 +276,52 @@ typedef struct st_client_loop_cb_t {
     struct sockaddr_storage server_address;
     struct sockaddr_storage client_address;
     struct sockaddr_storage client_alt_address;
+    int client_alt_if;
     picoquic_connection_id_t server_cid_before_migration;
     picoquic_connection_id_t client_cid_before_migration;
 } client_loop_cb_t;
 
+#ifdef _WINDOWS
+char *strsep(char **stringp, const char *delim) {
+    if (*stringp == NULL) {
+        return NULL;
+    }
+    char *token_start = *stringp;
+    *stringp = strpbrk(token_start, delim);
+    if (*stringp) {
+        **stringp = '\0';
+        (*stringp)++;
+    }
+    return token_start;
+}
+#endif
+
+/*
+ * mp_config is consisted with src_if and alt_ip, seperated by ","
+ * where src_if is an int, and alt_ip is a string.
+ * alt_ip is the ip of the alternative path
+ * src_if is the index of the interface where the alt_ip is bounded with
+ */
+int picoquic_parse_client_multipath_config(char *mp_config, int *src_if, struct sockaddr_storage *alt_ip)
```

Comment:
Is there any better way? I really don't want to introduce a self-implemented `strsep` for Windows.

Reply:
I am not sure. I am looking at the textual representation of IP addresses and interfaces on different systems. These are commonly used with IPv6 link local addresses. On windows, I see for example:
```
Wireless LAN adapter Wi-Fi:
   Link-local IPv6 Address . . . . . : fe80::e019:8019:6e0c:695c%16
   IPv4 Address. . . . . . . . . . . : 192.168.1.106
```

But on Mac/OS I see something like
```
llw0:
   inet6 fe80::e019:8019:6e0c:695c%llw0 prefixlen 64 scopeid 0xf
```
So, sadly, there is no standard. I would not have picked "," as a separator, but the only strong rule is that we must not use ".", ":" or "&". Maybe "/". Also, it might be a bit more conventional to write the interface index after the address than before it. Something like:
```
192.168.1.106/16
```
or
```
2001:23:45::ead5:6789:def0/16
```

Reply:
Alright, I changed it to '/'.